### PR TITLE
Add tests for xmagics, os, xutils, xinterpreter

### PR DESCRIPTION
**Pull request by tharun571**

### CODE REVIEW

**Comment by mcbarton** [comment]
sys/wait.h is for unix systems only. You will need a different header for Windows.

**Comment by mcbarton** [comment]
unistd.h is also unix only.

**Comment by tharun571** [comment]
@mcbarton the handler function which has `backtrace` and `backtrace_symbols_fd` are also not available in windows. Shall I change that particular test for unix only?

**Comment by mcbarton** [comment]
@tharun571 Maybe you could try one of the suggestions from this stackoverflow page where they talk about Windows equivalents https://stackoverflow.com/questions/341817/is-there-a-replacement-for-unistd-h-for-windows-visual-c .

**Comment by tharun571** [comment]
Also is there a way to check windows build in local? or should I push the code everytime to check?

**Comment by mcbarton** [comment]
> Also is there a way to check windows build in local? or should I push the code everytime to check?

If you have access to a Windows machine you could run the commands in the ci. There are currently no Windows instructions in the documentation. I will add at some point.

File: test/test_interpreter.cpp
Comment (anutosh491) [inline test/test_interpreter.cpp:178]: Do you intend to test the `complete_request` or the `is_complete_request` here

**Comment by anutosh491** [comment]
Converting to draft, let us know when the PR is ready for review (after maybe squashing the commits too)

**Comment by tharun571** [comment]
@anutosh491 this covers most of #114. Having some trouble with complete_request. Will push that later.

File: test/test_interpreter.cpp
Comment (tharun571) [reply test/test_interpreter.cpp:178]: `is_complete_request`

Comment (anutosh491) [reply test/test_interpreter.cpp:176]: Works.

**Review by anutosh491 — APPROVED** [review]
Looks good. Let's wait for maybe one more review before we merge this.
cc @mcbarton

**Pull request merged**
